subsys/bluetooth/mesh/light_ctl_srv.c

```diff
@@ -0,0 +1,618 @@
+#include <bluetooth/mesh/gen_dtt_srv.h>
+#include "light_ctl_internal.h"
+#include "lightness_internal.h"
+#include "model_utils.h"
+#define BT_DBG_ENABLED IS_ENABLED(CONFIG_BT_MESH_DEBUG_MODEL)
+#define LOG_MODULE_NAME bt_mesh_light_ctl_srv
+#include "common/log.h"
+
+struct bt_mesh_light_ctl_srv_settings_data {
+	struct bt_mesh_light_ctl_params default_params;
+	struct bt_mesh_light_ctl_temp_range temp_range;
+	uint16_t temp_last;
+} __packed;
+
+static int store_state(struct bt_mesh_light_ctl_srv *srv)
+{
+	if (!IS_ENABLED(CONFIG_BT_SETTINGS)) {
+		return 0;
+	}
+
+	struct bt_mesh_light_ctl_srv_settings_data data = {
+		.default_params = srv->default_params,
+		.temp_range = srv->temp_srv.temp_range,
+		.temp_last = srv->temp_srv.temp_last,
+
+	};
+
+	return bt_mesh_model_data_store(srv->model, false, NULL, &data,
+					sizeof(data));
+}
+
+static void ctl_encode_status(struct net_buf_simple *buf,
+			      struct bt_mesh_light_ctl_status *status)
+{
+	bt_mesh_model_msg_init(buf, BT_MESH_LIGHT_CTL_STATUS);
+	net_buf_simple_add_le16(buf, status->current_light);
+	net_buf_simple_add_le16(buf, status->current_temp);
+
+	if (status->remaining_time != 0) {
+		net_buf_simple_add_le16(buf, status->target_light);
+		net_buf_simple_add_le16(buf, status->target_temp);
+		net_buf_simple_add_u8(
+			buf, model_transition_encode(status->remaining_time));
+	}
+}
+
+static void ctl_rsp(struct bt_mesh_model *model, struct bt_mesh_msg_ctx *rx_ctx,
+		    struct bt_mesh_light_ctl_gen_cb_status *gen_status)
+{
+	BT_MESH_MODEL_BUF_DEFINE(msg, BT_MESH_LIGHT_CTL_STATUS,
+				 BT_MESH_LIGHT_CTL_MSG_MAXLEN_STATUS);
+	struct bt_mesh_light_ctl_status status = {
+		.current_light = gen_status->current.light,
+		.current_temp = gen_status->current.temp,
+		.target_light = gen_status->target.light,
+		.target_temp = gen_status->target.temp,
+		.remaining_time = gen_status->remaining_time,
+	};
+	ctl_encode_status(&msg, &status);
+	(void)bt_mesh_model_send(model, rx_ctx, &msg, NULL, NULL);
+}
+
+static void light_ctl_pub(struct bt_mesh_light_ctl_srv *srv,
+			  struct bt_mesh_light_ctl_gen_cb_status *gen_status)
+{
+	struct bt_mesh_light_ctl_status status = {
+		.current_light = gen_status->current.light,
+		.current_temp = gen_status->current.temp,
+		.target_light = gen_status->target.light,
+		.target_temp = gen_status->target.temp,
+		.remaining_time = gen_status->remaining_time,
+	};
+
+	(void)bt_mesh_light_ctl_pub(srv, NULL, &status);
+}
+
+static void ctl_set(struct bt_mesh_model *model, struct bt_mesh_msg_ctx *ctx,
+		    struct net_buf_simple *buf, bool ack)
+{
+	if (buf->len != BT_MESH_LIGHT_CTL_MSG_MINLEN_SET &&
+	    buf->len != BT_MESH_LIGHT_CTL_MSG_MAXLEN_SET) {
+		return;
+	}
+
+	struct bt_mesh_light_ctl_srv *srv = model->user_data;
+	struct bt_mesh_light_ctl_gen_cb_status status;
+	struct bt_mesh_light_ctl_gen_cb_set set;
+	struct bt_mesh_model_transition transition;
+	uint16_t light = net_buf_simple_pull_le16(buf);
```

Comment:
This needs to be converted from light_actual to light_repr using the subsys/lightness.h utils. Same for status sending and the default CTL set/get